com.liferay.blade.upgrade.liferay70/src/com/liferay/blade/upgrade/liferay70/apichanges/IndexerThreeMethodsChange.java

```diff
@@ -48,15 +48,20 @@
 		final List<SearchResult> searchResults = new ArrayList<>();
 
 		List<SearchResult> declarations = javaFileChecker.findMethodDeclaration(
-				"addRelatedEntryFields", new String[] { "Document", "Object" }, null);
+				"addRelatedEntryFields", new String[] { "Document", "Object" }, null,
```

Comment:
Lets do something like this
```
List<SearchResult> interfaceSearch = javaFileChecker.findImplementsInterface("RelatedEntryIndexer");

		if (interfaceSearch.isEmpty()) {
			List<SearchResult> declarations = javaFileChecker.findMethodDeclaration(
					"addRelatedEntryFields", new String[] { "Document", "Object" }, null);
			searchResults.addAll(declarations);
		}
```

Reply:
This way we should not need to change the JavaFile interface or add new ICondition service.